Start documentation

lukaspie opened the pull request.

mkuehbach reviewed (commented):
That is quite a sophisticated start @lukaspie, I have not much to comment here, mainly focusing on other work at the moment some points catched my thoughts:
- You have currently some documentation still on the repo's README.md and some already in the doc, do you plan to remove most in the README.md cuz single source principle could be the README is just the link to the doc and maybe here its getting started page
- I like that you picked up some of the work and thoughts from EM I also like the individual acknowledgements for examples in the reference page

To me this reads ready for a first release as it is meaty enough to be discussed by XPS experts and the community

lukaspie commented on docs/explanation/contextualization.md:
```diff
@@ -0,0 +1,8 @@
+# How to map pieces of information to NeXus
+
+<!-- Conceptually, mapping between representations of concepts and instance data is a key tasks in information science. The plugin pynxtools-em implements this specifically for the file and serialization formats used within the research field of electron microscopy. The plugin offloads this code from eventually other software where NeXus is used to avoid that every user and software developer has ...
```
I changed this file completely, this was just a placeholder. Thanks!

sanbrock commented on docs/index.md:
```diff
@@ -0,0 +1,69 @@
+
+### Reference
+
+Here you can learn which specific measurement setups and file formats from technology partners pynxtools-xps currently supports.
```
This can be the place for the full list of currently supported formats. I suggest to detail the short name or file extensions of the different data types for each already here, so without the need of clicking one can the the full picture.  

Actually, this could also go to have a generic pynxtools reference page which lists all supported file formats and that which registered plugin(s) implements them.  

There could also be a reader method which returns such documentation, so a given pynxtools installation could not only list the available readers, but also the file formats they support

lukaspie replied:
I added a list of currently supported formats here now.

I think your last two points will be especially relevant once we use the multi-format reader consistenly across the different plugins. Then we can directly support which file formats are supported. I have created two separate issues (https://github.com/FAIRmat-NFDI/pynxtools/issues/320, https://github.com/FAIRmat-NFDI/pynxtools/issues/321) in pynxtools for this.

lukaspie commented:
> * You have currently some documentation still on the repo's README.md and some already in the doc, do you plan to remove most in the README.md cuz single source principle could be the README is just the link to the doc and maybe here its getting started page

I removed some parts from the README now and replaced them by a link to the docs. For the individual examples, I kept the README, so people can see them immediately in the repo as well. Maybe eventually the docs could also just usethe README files to populate the docs (or vice-versa).
